src/app/shared/services/http-v2/http-v2.service.ts

```diff
@@ -76,13 +78,15 @@ export class HttpV2Service {
     data: any = {},
     responseClass?: ResponseClass<T>,
     options: RequestOptions = defaultOptions,
+    responseType?: 'json' | 'text',
```

Comment:
Since we're using this `'json' | 'text'` typing everywhere, we should probably just define it as a specific type like
```ts
type ResponseType = 'json' | 'text';
```

That way if we ever have to communicate with an external API that uses XML for instance we only have to change this typing on one line instead of many.